adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -93,13 +95,29 @@ - (void)initWithAppId:(nonnull NSString *)appId
   }
 
   _isInitializing = YES;
-  _isBannerPresenting = NO;
 
   // Disable refresh functionality for all banners
   [[VungleSDK sharedSDK] disableBannerRefresh];
 
+  // Set init options for priority placement
+  NSMutableDictionary *initOptions = [NSMutableDictionary dictionary];
+  if (delegate) {
+    NSString *priorityPlacementID = delegate.desiredPlacement;
+    [initOptions setObject:priorityPlacementID forKey:VungleSDKInitOptionKeyPriorityPlacementID];
+    _prioritizedPlacementID = [priorityPlacementID copy];
```

Comment:
Don't need copy if the property itself becomes copy.

Reply:
@ericleich Hi Eric, we only defined an instance variable `_prioritizedPlacementID`, we didn't define a property `prioritizedPlacementID`, so we add copy method here. Thanks a lot. ^_^

Reply:
Ah, so `@property(nonatomic, nonnull) NSString *desiredPlacement;` on GADMAdapterVungleDelegate should also be copy. That way, way, you can just write:

```
_prioritizedPlacementID = delegate.desiredPlacement
```

And you will get a copy by default as you're accessing the desiredPlacement property.

Reply:
@ericleich Hi Eric, I have a concern that the `copy` attribute is only for setter method, not for getter method:

Excerpt from stackoverflow -----
The `copy` attribute means only that the setter will take a copy when the property is set. The synthesized getter will not return a copy. You must implement that behavior yourself:
https://stackoverflow.com/questions/25111218/do-i-need-to-override-the-getter-for-a-property-to-return-an-immutable-copy

Reply:
Hmm, but I think NSString behaves differently is different?

I ran this test:

```
  NSString *a = @"a";
  NSString *other = a;
  a = @"b";
  NSLog(@"Other string: %@", other);
```

and it logged;

```
Other string: a
```

Reply:
Yeah, NSString is special. I also ran this test:

```
  GADInterstitial *interstitial = [[GADInterstitial alloc] initWithAdUnitID:@"abc"];
  GADInterstitial *otherInterstitial = interstitial;
  interstitial.adUnitID = @"def";
  NSLog(@"Other interstitial ad unit: %@", otherInterstitial.adUnitID);
```

and got:

```
Other interstitial ad unit: def
```

So while pointers are typically soft copy, NSString seems to make a deep copy on get.

Reply:
If you feel strongly about this, I can accept. But it's quite possible we may do an internal refactor later. I haven't seen code needing to copy strings (other than setter property being copy), and if that were the case, I feel like I'd see it everywhere.

As another example, your code doesn't pass a copy to initOptions. Wouldn't you want to make a copy there if we were doing copies of strings?

Reply:
@ericleich Hi Eric, thanks so much for showing me the code snippets,  I will update the code according to your instructions. ^_^

Reply:
@ericleich Updated, thanks a lot. 😊